### PR TITLE
fixes failing snapshots because timeout is too short

### PR DESCRIPTION
**Pull request by ebostijancic**
## Description

If the cloud provider snapshot takes longer than 10minutes, terraform apply will fail. This pull request make the timeout configurable.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the Terraform contribution guidelines
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go]: [q] what about the scenarios where `timeout` is set to a negative number or empty string?
[q] why `timeout` is a string and not an interger?

**Review by andreaangiolillo — APPROVED** [review]
LGTM with a question

**Comment by ebostijancic** [comment]
A time duration cannot be negative and it's string because it's defined like: "1h" or "30m". I admit it would be better to be declared as a block see operational timeouts for more information: 

[Terraform Operation Timeouts](https://www.terraform.io/docs/language/resources/syntax.html#operation-timeouts)

File: mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go
Comment (themantissa) [inline mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go]: If we are including a new option it needs to be added to the documentation as well for the provider.

**Review by themantissa — CHANGES_REQUESTED** [review]
See comment on docs. Please include that. Thank you!

**Comment by themantissa** [comment]
@ebostijancic thank you for the submission.   Did you sign the CLA yet?  https://www.mongodb.com/legal/contributor-agreement   We will need that completed before we can merge in the work, once approved.  Thank you!

**Comment by ebostijancic** [comment]
> @ebostijancic thank you for the submission. Did you sign the CLA yet? https://www.mongodb.com/legal/contributor-agreement We will need that completed before we can merge in the work, once approved. Thank you!

I have signed the CLA and updated the documention

File: mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go
Comment (ebostijancic) [reply mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go]: documentation can be found in the followup commit.

File: mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go
Comment (thetonymaster) [inline mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go]: Add the default `10m` string on the `Default` field

File: website/docs/r/cloud_provider_snapshot.html.markdown
Comment (thetonymaster) [inline website/docs/r/cloud_provider_snapshot.html.markdown]: Can we also add a little explanation as in how the duration is supposed to be written? For Go people this can be self-explanatory but for non-go people this can be hard to understand.

**Comment by themantissa** [comment]
@ebostijancic thank you for signing the CLA!  Once you respond to @thetonymaster  and have his approval we can move this PR forward.

**Review by themantissa — APPROVED** [review]
LGTM, just needs comments from @thetonymaster  addressed.

**Review by thetonymaster — APPROVED** [review]
LGTM

**Comment by themantissa** [comment]
@ebostijancic and @thetonymaster fyi the tests are failing with
 --- FAIL: TestProvider (0.01s)
    provider_test.go:44: err: 1 error occurred:
        	* resource mongodbatlas_cloud_provider_snapshot: No Update defined, must set ForceNew on: []string{"timeout"}

File: mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go
Comment (coderGo93) [inline mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go]: It should be optional, shouldn't it? and it must have `ForceNew: True` since it doesn't have an update.

**Review by coderGo93 — CHANGES_REQUESTED** [review]
Looking good but realized some small mistakes like `timeout` is required when it should be optional and lacking the ForceNew

**Comment by thetonymaster** [comment]
> @ebostijancic and @thetonymaster fyi the tests are failing with --- FAIL: TestProvider (0.01s) provider_test.go:44: err: 1 error occurred: * resource mongodbatlas_cloud_provider_snapshot: No Update defined, must set ForceNew on: []string{"timeout"}

Yeah the tests didn't ran for some reason until I started the job

**Comment by themantissa** [comment]
@ebostijancic wanted to check in here and see if you've seen the most recent comments and test failure?

File: mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go
Comment (ebostijancic) [reply mongodbatlas/resource_mongodbatlas_cloud_provider_snapshot.go]: please check latest commit.

**Comment by themantissa** [comment]
@ebostijancic  once we have a sign off form @coderGo93  we'll merge this.  Thank you!

**Review by coderGo93 — APPROVED** [review]
LGTM, thank you

**Pull request merged**
